### PR TITLE
add docs to pipeline run API types

### PR DESCRIPTION
**Pull request by anbrsap**

### CODE REVIEW

File: pkg/apis/steward/v1alpha1/run_types.go [hunk 1]
```diff
@@ -142,12 +195,24 @@ const (
...
-	JobName string `json:"jobName"`
-	// SequenceNumber is a sequential number of the run
-	SequenceNumber int `json:"sequenceNumber"`
-	// Cause is the cause which triggered the run, e.g. a SCM change, an user action or a timer.
-	Cause string `json:"cause"`
+
+	// JobName is the name of the job this pipeline run belongs to. It is used
+	// as the name of the Jenkins job and therefore must be a valid Jenkins job
+	// name. If empty, a default name will be used for the Jenkins job.
+	// +optional
```
Comment (alxsap) [inline pkg/apis/steward/v1alpha1/run_types.go]:
```suggestion
	// TODO: Regex in CRD to validate "valid Jenkins job names"
	// +optional
```

File: pkg/apis/steward/v1alpha1/run_types.go [hunk 2]
```diff
@@ -142,12 +195,24 @@ const (
...
+	JobName string `json:"jobName,omitempty"`
+
+	// SequenceNumber is the sequence number of the pipeline run, which
+	// translates into the build number of the Jenkins job.
+	// +optional
+	SequenceNumber int `json:"sequenceNumber,omitempty"`
+
+	// Cause a textual description of the cause of this pipeline run. Will be
```
Comment (alxsap) [inline pkg/apis/steward/v1alpha1/run_types.go]:
```suggestion
	// Cause is a textual description of the cause of this pipeline run. Will be
```

File: pkg/apis/steward/v1alpha1/run_types.go [hunk 3]
```diff
@@ -142,12 +195,24 @@ const (
 	IntentAbort Intent = "abort"
 )
 
-// PipelineRunDetails provies details of a pipeline run which are passed to the jenkinsfile-runner.
+// PipelineRunDetails provides details of a pipeline run which are passed to
```
Comment (alxsap) [inline pkg/apis/steward/v1alpha1/run_types.go]:
```suggestion
// PipelineRunDetails provides metadata for a pipeline run which is evaluated by
```

File: pkg/apis/steward/v1alpha1/run_types.go [hunk 4]
```diff
@@ -28,37 +32,83 @@ type PipelineRunList struct {
...
+	// to be executed.
 	JenkinsFile JenkinsFile `json:"jenkinsFile"`
+
+	// Args contains the key-value parameters to pass to the pipeline.
 	// +optional
 	Args map[string]string `json:"args"`
+
+	// Secrets is the list of secrets to be made available to the pipeline
+	// execution. Each entry in the list is the name of a Kubernetes `v1/Secret`
+	// resource object in the same namespace as the PipelineRun object itself.
```
[the remark this answers is not in the document]
Comment (anbrsap) [reply pkg/apis/steward/v1alpha1/run_types.go]: Actually this is an implementation detail. I think its enough to say that the secrets are "to be made available to the pipeline execution".

File: pkg/apis/steward/v1alpha1/run_types.go [hunk 5]
```diff
@@ -28,37 +32,83 @@ type PipelineRunList struct {
...
 	// +optional
 	Secrets []string `json:"secrets"`
+
+	// ImagePullSecrets is the list of image pull secrets required by the
+	// pipeline run to pull images of custom containers from private registries.
+	// Each entry in the list is the name of a Kubernetes `v1/Secret` resource
+	// object of type `kubernetes.io/dockerconfigjson` in the same namespace as
+	// the PipelineRun object itself.
```
[the remark this answers is not in the document]
Comment (anbrsap) [reply pkg/apis/steward/v1alpha1/run_types.go]: Dito.

File: pkg/apis/steward/v1alpha1/run_types.go [hunk 6]
```diff
@@ -28,37 +32,83 @@ type PipelineRunList struct {
...
 // JenkinsFile represents the location from where to get the pipeline
 type JenkinsFile struct {
-	URL      string `json:"repoUrl"`
+
+	// URL is the URL of the Git repository containing the pipeline definition
+	// (aka `Jenkinsfile`).
+	URL string `json:"repoUrl"`
+
+	// Revision is the revision of the pipeline Git repository to used, e.g.
+	// `master`.
```
[the remark this answers is not in the document]
Comment (anbrsap) [reply pkg/apis/steward/v1alpha1/run_types.go]: Defaulting can be done by the client (if the end user does not specify the revision, the client passes `master` here).

File: pkg/apis/steward/v1alpha1/run_types.go [hunk 7]
```diff
@@ -28,37 +32,83 @@ type PipelineRunList struct {
...
 	Revision string `json:"revision"`
-	Path     string `json:"relativePath"`
+
+	// Path is the relative pathname of the pipeline definition file in the
+	// repository check-out, typically `Jenkinsfile`.
+	Path string `json:"relativePath"`
```
[the remark this answers is not in the document]
Comment (anbrsap) [reply pkg/apis/steward/v1alpha1/run_types.go]: Dito.

File: pkg/apis/steward/v1alpha1/run_types.go (hunk 1 above)
Comment (anbrsap) [reply pkg/apis/steward/v1alpha1/run_types.go]: Good idea!

**Pull request merged**